signer/src/network/libp2p/bootstrap.rs

```diff
@@ -47,6 +51,19 @@ impl Config {
         self
     }
 
+    /// Adds known peers to the configuration.
+    pub fn with_known_peers<'a, T>(mut self, known_peers: T) -> Self
+    where
+        T: IntoIterator<Item = &'a (PeerId, Multiaddr)>,
```

Comment:
Why take a reference when we actually want ownership? Perhaps the caller has an iterator of owned items to feed in here, in which case they can give it up. If they don't then they can do the cloning.

Also, we could just require a `Vec`, since this function is called with `Vec`s.